luaV_execute.c

```diff
@@ -153,64 +67,64 @@ static const void *const disptab[((int)(OP_EXTRAARG) + 1)] = {
     ((void)0);
 
     ((void)0);
-    goto *disptab[(((OpCode)(((i)>>0) & ((~((~(Instruction)0)<<(7)))<<(0)))))]; {
-      L_OP_MOVE: {
+    switch((((OpCode)(((i)>>0) & ((~((~(Instruction)0)<<(7)))<<(0)))))) {
+      case OP_MOVE: {
         { TValue *io1=((&(ra)->val)); const TValue *io2=((&((base+((((int)((((i)>>((((0 + 7) + 8) + 1))) & ((~((~(Instruction)0)<<(8)))<<(0)))))))))->val)); io1->value_ = io2->value_; ((io1)->tt_=(io2->tt_)); ((void)L, ((void)0)); ((void)0); };
-        { if (trap) { trap = luaG_traceexec(L, pc); (base = ci->func + 1); } i = *(pc++); ra = (base+(((int)((((i)>>((0 + 7))) & ((~((~(Instruction)0)<<(8)))<<(0))))))); }; goto *disptab[(((OpCode)(((i)>>0) & ((~((~(Instruction)0)<<(7)))<<(0)))))];;;
+        break;
```

Comment:
My main concern is that the behaviour might differ, here, but the loop prologue looks equivalent.